docs/adr/adr-006-fraud-service.md

```diff
@@ -137,26 +141,19 @@ type Subscription interface {
    Proof() (Proof, error)
    Cancel() error
 }
-
-// FraudSub implements Subscription.
-type FraudSub struct {
-   pubsub *pubsub.PubSub 
-}
-
-func NewFraudSub(p *pubsub.PubSub)(Subscription, error){}
-func(s *FraudSub) Proof() (Proof, error){}
-func(s *FraudSub) Cancel() error{}
 ```
 
 ```go
-type FraudService struct {
-   fraudSub Subscription
-   fraudNotifier FraudNotifier
+// FraudSub implements Subscriber and Broadcaster.
+type FraudSub struct {
+   pubsub *pubsub.PubSub
+   topics map[ProofType]*pubsub.Topic
+   unmarshallers map[ProofType]proofUnmarshaller
 }
 
-func(s *FraudService) Proof() (Proof, error){}
-func(s *FraudSub) Cancel() error{}
-func(f *FraudService) Notify(ctx context.Context, p Proof){}
+func(s *FraudService) RegisterUnmarshaller(proofType ProofType, f proofUnmarshaller) error{}
+func(s *FraudService) Subscribe(ctx context.Context, proofType ProofType) (Subscription, error){}
+func(s *FraudService) Broadcast(ctx context.Context, p Proof){}
```

Comment:
How will it know which topic to Broadcast it to? You have to type check on the given proof?

Reply:
by calling `p.Type()` inside.